Game/Source/GDKTestGyms/Private/DeterministicBlackboardValues.cpp

```diff
@@ -0,0 +1,81 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "DeterministicBlackboardValues.h"
+
+#include "AIController.h"
+#include "BehaviorTree/BlackboardComponent.h"
+#include "NavigationSystem.h"
+
+DEFINE_LOG_CATEGORY(LogDeterministicBlackboardValues);
+
+// Sets default values for this component's properties
+UDeterministicBlackboardValues::UDeterministicBlackboardValues()
+{
+	bWantsInitializeComponent = true;
+	//PrimaryComponentTick.bWantsInitializeComponent = true;
+}
+
+void UDeterministicBlackboardValues::InitializeComponent() 
+{
+	Super::InitializeComponent();
+}
+
+// Called every frame
+void UDeterministicBlackboardValues::TickComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction)
+{
+	Super::TickComponent(DeltaTime, TickType, ThisTickFunction);
+}
+
+void UDeterministicBlackboardValues::ApplyValues()
+{
+	APawn* P = Cast<APawn>(GetOwner());
+	AController* Controller = P->GetController();
+
+	static FName TargetA{ TEXT("TargetA") };
+	static FName TargetB{ TEXT("TargetB") };
+
+	AAIController* AIController = Cast<AAIController>(Controller);
+	if (AIController == nullptr)
+	{
+		AIController = Cast<AAIController>(P);
+	}
+	if (AIController)
+	{
+		UBlackboardComponent* Blackboard = (UBlackboardComponent*)AIController->GetBlackboardComponent();
+		checkf(Blackboard, TEXT("AI Controller did not have a blackboard %s"), *Controller->GetPawn()->GetName());
+
+		FVector WorldLocation = ((AActor*)Controller)->GetActorLocation(); // GetWorldLocation?
+
+	// 	FVector RunLocationA = WorldLocation + Locations.A;
+	// 	FVector RunLocationB = WorldLocation + Locations.B;
+
+		UNavigationSystemV1* NavSys = UNavigationSystemV1::GetCurrent(GetWorld());
+
+		FNavLocation LocA;
+		checkf(NavSys->GetRandomPointInNavigableRadius(WorldLocation + RunPointA, 100.0f, LocA), TEXT("Was not able to find point to nav to."));
```

Comment:
Can't put this in a check or it'll get optimised away in none-checking builds.
Also 100.f seems a little magic.

Reply:
It's totally magic, I added some tolerance to allow for some distance between the point and the nav mesh surface - 1m seemed to do the trick without breaking the determinism